tests/import_schemas.py

```diff
@@ -0,0 +1,213 @@
+                    ret.append((field.name, field.type.name))
+                elif isinstance(field.type, avro.schema.UnionSchema):
+                    t0 = field.type.schemas[0]
+                    t1 = field.type.schemas[1]
+                    if (isinstance(t0, avro.schema.PrimitiveSchema) and
+                            t0.type == "null"):
+                        if isinstance(t1, avro.schema.RecordSchema):
+                            ret.append((field.name, t1.name))
+                    else:
+                        msg = "Schema union assumptions violated: {}.{}"
+                        raise Exception(msg.format(self.name, field.name))
+        return ret
+
+
+class SchemaGenerator(object):
+    """
+    Class that generates a schema in Python code from Avro definitions.
+    """
+    def __init__(self, version, schemaDir, verbosity):
+        self.version = version
+        self.schemaDir = schemaDir
+        self.verbosity = verbosity
+        self.classes = []
+        for avscFile in glob.glob(os.path.join(self.schemaDir, "*.avsc")):
+            self.classes.append(SchemaClass(avscFile))
+        self.requestClassNames = [
+            cls.name for cls in self.classes if cls.isSearchRequest()]
+        self.responseClassNames = [
+            cls.name for cls in self.classes if cls.isSearchResponse()]
+        self.postSignatures = []
+        for request, response in zip(
+                self.requestClassNames, self.responseClassNames):
+            objname = re.search('Search(.+)Request', request).groups()[0]
+            url = '/{0}/search'.format(objname.lower())
+            tup = (url, request, response)
+            self.postSignatures.append(tup)
+        self.postSignatures.sort()
+
+    def getClasses(self):
+        return self.classes
+
+
+class SchemaProcessor(object):
+    """
+    Transforms schemas into Python objects
+    """
+    def __init__(self, args):
+        self.version = args.version
+        self.verbosity = args.verbose
+        self.tmpDir = tempfile.mkdtemp(prefix="ga4gh_")
+        self.sourceTar = os.path.join(self.tmpDir, "schemas.tar.gz")
+        self.avroJarPath = args.avro_tools_jar
+        # Note! The tarball does not contain the leading v
+        string = "schemas-{0}".format(self.version[1:])
+        self.schemaDir = os.path.join(self.tmpDir, string)
+        self.avroJar = os.path.join(self.schemaDir, "avro-tools.jar")
+        self.sourceDir = args.inputSchemasDirectory
+        self.avroPath = "src/main/resources/avro"
+
+    def cleanup(self):
+        if self.verbosity > 1:
+            utils.log("Cleaning up tmp dir {}".format(self.tmpDir))
+        shutil.rmtree(self.tmpDir)
+
+    def download(self, url, destination):
+        """
+        Downloads the specified url and saves the result to the specified
+        file.
+        """
+        fileDownloader = utils.FileDownloader(url, destination)
+        fileDownloader.download()
+
+    def convertAvro(self, avdlFile):
```

Comment:
I don't think we can; if it's not done here, converting to avsc will have to be done in the testing script.